clippy_lints/src/doc/mod.rs

```diff
@@ -658,9 +658,7 @@ impl<'tcx> LateLintPass<'tcx> for Documentation {
                 );
                 match item.kind {
                     ItemKind::Fn { sig, body: body_id, .. } => {
-                        if !(is_entrypoint_fn(cx, item.owner_id.to_def_id())
-                            || item.span.in_external_macro(cx.tcx.sess.source_map()))
-                        {
+                        if !(item.span.in_external_macro(cx.tcx.sess.source_map())) {
```

Comment:
seems the extra parentheses `()` is unnecessary here.

```suggestion
                        if !item.span.in_external_macro(cx.tcx.sess.source_map()) {
```